dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -14,15 +14,20 @@ import com.datadog.android.rum.RumMonitor
 import com.datadog.android.rum.RumResourceKind
 import com.datadog.android.rum.internal.domain.RumEvent
 import com.datadog.android.rum.internal.domain.RumEventData
+import com.datadog.android.rum.internal.domain.RumEventSerializer
+import java.lang.ref.WeakReference
 import java.util.UUID
 
 internal class DatadogRumMonitor(
     private val timeProvider: TimeProvider,
     private val writer: Writer<RumEvent>
 ) : RumMonitor {
 
-    private val activeView = mutableMapOf<Any, RumEvent>()
-    private val activeResources = mutableMapOf<Any, RumEvent>()
+    private var activeViewEvent: RumEvent? = null
+    private var activeViewData: RumEventData.View? = null
+    private var activeViewKey: WeakReference<Any?> = WeakReference(null)
```

Comment:
This approach will not work for the case where you have 2 active activities :(. I think we need to re - think this, using the WeakHashMap will work I think. The WeakHashMap will automatically clean itself whenever a resource/view is not strongly referenced anymore and the user didn't close it itself. 

The other approach will be the one used for the "activeResources" but here we can go even further and use a ReferenceQueue in the WeakReference initialization. Every time there's a new startView/stopView called we check the ReferenceQueue and close all the views that were unreferenced and never closed.

Reply:
When would you have 2 active activities ? The idea is that there's only one single view active at a single time.

Reply:
There's only one corner case but yeah...we can add this later if someone complains about it.